farcaster-frame/contracts/evm/solidity/CanvasGame.sol

```diff
@@ -33,34 +39,36 @@ contract CanvasGame is Ownable {
     }
 
     function paint(uint256 canvas, uint24 color) public payable {
-        require(msg.value >= fee, "Sufficient fee required to paint");
-        require(canvas < canvasOwner.length, "Canvas ID must exist");
-        require(paintCount[canvas] < paintLimit, "Canvas must not be full");
+        if (!(msg.value >= fee)) revert InsufficientFee();//msg.value, fee); // Sufficient fee required to paint
+        if (!(canvas < canvasOwner.length)) revert CanvasDoesNotExist();//canvas, canvasOwner.length); // Canvas ID must exist
+        if (!(paintCount[canvas] < paintLimit)) revert CanvasFull();//canvas, paintLimit); // Canvas must not be full
 
         uint256 contractOwnerReward = msg.value / 10;
         uint256 canvasOwnerReward = msg.value - contractOwnerReward;
         rewards[owner()] += contractOwnerReward;
         rewards[canvasOwner[canvas]] += canvasOwnerReward;
 
         painted[canvas][msg.sender] = true;
-        paintCount[canvas] += 1;
+        ++paintCount[canvas];
 
         emit Paint(msg.sender, canvas, color);
     }
 
     function fork(uint256 canvas) public {
-        require(canvas < canvasOwner.length, "Canvas ID must exist");
-        require(painted[canvas][msg.sender], "Must have painted to this canvas before");
-        require(canvasOwner[canvas] != msg.sender, "Must not fork your own canvas");
-        require(!forked[canvas][msg.sender], "Must not have forked this canvas already");
+        if (!(canvas < canvasOwner.length)) revert CanvasDoesNotExist();//canvas, canvasOwner.length); // Canvas ID must exist
+        if (!(painted[canvas][msg.sender])) revert InvalidFork();//canvas); // Must not have painted to this canvas before
+        if (!(canvasOwner[canvas] != msg.sender)) revert InvalidFork();//canvas); // Must not fork your own canvas
+        if (!(!forked[canvas][msg.sender])) revert InvalidFork();//canvas); // Must not have forked this canvas already
```

Comment:
Please fix the double negatives in condition 3 and 4, and flip the condition from `!(x < y)` to `x >= y` in the condition 1